frontend/src/employee-frontend/components/child-information/placements/NewServiceNeeds.tsx

```diff
@@ -36,28 +83,237 @@ function NewServiceNeeds({ serviceNeeds }: Props) {
+  onEdit,
+  onDelete,
+  disabled
+}: NewServiceNeedReadRowProps) {
+  const { i18n } = useTranslation()
+  return (
+    <Tr>
+      <Td>
+        {serviceNeed.startDate.format()} - {serviceNeed.endDate.format()}
+      </Td>
+      <Td>{serviceNeed.option.name}</Td>
+      <Td>{serviceNeed.shiftCare ? i18n.common.yes : i18n.common.no}</Td>
+      <Td>
+        <Toolbar
+          dateRange={serviceNeed}
+          onEdit={onEdit}
+          editableFor={['ADMIN', 'UNIT_SUPERVISOR']}
+          onDelete={onDelete}
+          deletableFor={['ADMIN', 'UNIT_SUPERVISOR']}
+          disableAll={disabled}
+        />
+      </Td>
+    </Tr>
+  )
+}
+
+interface FormData {
+  startDate: LocalDate | undefined
+  endDate: LocalDate | undefined
+  optionId: UUID | undefined
+  shiftCare: boolean
+}
+
+interface NewServiceNeedCreateRowProps {
+  placement: Placement
+  options: DropdownOption[]
+  initialForm: FormData
+  onSuccess: () => void
+  onCancel: () => void
+  editingId?: string
+}
+function NewServiceNeedEditorRow({
+  placement,
+  options,
+  initialForm,
+  onSuccess,
+  onCancel,
+  editingId
+}: NewServiceNeedCreateRowProps) {
+  const { i18n } = useTranslation()
+  const t = i18n.childInformation.placements.serviceNeeds
+
+  const { setErrorMessage } = useContext(UIContext)
+  const [form, setForm] = useState<FormData>(initialForm)
+  const [submitting, setSubmitting] = useState(false)
+
+  const formIsValid =
+    form.startDate &&
+    form.endDate &&
+    form.optionId &&
+    !form.endDate.isBefore(form.startDate)
+
+  const onSubmit = () => {
+    if (form.startDate && form.endDate && form.optionId) {
+      setSubmitting(true)
+
+      if (editingId) {
+        void updateNewServiceNeed(editingId, {
+          startDate: form.startDate,
+          endDate: form.endDate,
+          optionId: form.optionId,
+          shiftCare: form.shiftCare
+        })
+          .then((res) => {
+            if (res.isSuccess) {
+              onSuccess()
+            } else {
+              setErrorMessage({
+                type: 'error',
+                title: i18n.common.error.unknown,
+                text: i18n.common.error.saveFailed,
+                resolveLabel: i18n.common.ok
+              })
+            }
+          })
+          .finally(() => setSubmitting(false))
+      } else {
+        void createNewServiceNeed({
+          placementId: placement.id,
+          startDate: form.startDate,
+          endDate: form.endDate,
+          optionId: form.optionId,
+          shiftCare: form.shiftCare
+        })
+          .then((res) => {
+            if (res.isSuccess) {
+              onSuccess()
+            } else {
+              setErrorMessage({
+                type: 'error',
+                title: i18n.common.error.unknown,
+                text: i18n.common.error.saveFailed,
+                resolveLabel: i18n.common.ok
+              })
+            }
+          })
+          .finally(() => setSubmitting(false))
+      }
```

Comment:
Everything apart from the initial request is identical between the branches, so I would do this instead:
```
const request = editingId ? updateNewServiceNeed(...) : createNewServiceNeed(...)
void request.then(...).finally(...)
```